bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2982,6 +2981,9 @@ void RewriteInstance::selectFunctionsToProcess() {
     if (mustSkip(Function))
       return false;
 
+    if (opts::MatchProfileWithFunctionHash)
+      return true;
+
```

Comment:
Can we instead modify `YAMLProfileReader::mayHaveProfileData()` to return `true` when `opts::MatchProfileWithFunctionHash` is set?

Reply:
If this check is done in YAMLProfileReader::mayHaveProfileData(), it must also be done in DataAggregator::mayHaveProfileData(), so I think having the check in ShouldProcess makes sense from the perspective of code duplicity, as well as organizationally

Reply:
Why does it have to be done in DataAggregator::mayHaveProfileData()?

Reply:
`> if (ProfileReader && !ProfileReader->mayHaveProfileData(Function))
        return false;`

In the same lambda, ProfileReader's mayHaveProfileData is called, which could be either a YamlProfileReader or a DataAggregator

Reply:
But hashed-based matching is limited to YAML profiles, isn't it?

Reply:
You're right - I'll move it there